### PR TITLE
Add Paratest task

### PR DESCRIPTION
**Pull request by janvernieuwe**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any

A PR with only paratest, not all tasks in parallel, so we can get this merged faster

# New Task Checklist:

- [ x ] Is the README.md file updated?
- [ x ] Are the dependencies added to the composer.json suggestions?
- [ x ] Is the doc/tasks.md file updated?
- [ x ] Are the task parameters documented?
- [ x ] Is the task registered in the tasks.yml file?
- [ x ] Does the task contains phpspec tests?
- [ x ] Is the configuration having logical allowed types?
- [ x ] Does the task run in the correct context?
- [ x ] Is the `run()` method readable?
- [ x ] Is the `run()` method using the configuration correctly?
- [ x ] Are all CI services returning green?

### CODE REVIEW

File: doc/tasks/paratest.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# Phpunit
```
Comment (veewee) [inline doc/tasks/paratest.md]: paratest

File: doc/tasks/paratest.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            log-junit: null
+            testsuite: null
+```
+
+**processes**
+
+*Default: null*
+
+The number of test processes to run. (Default: auto)
+                             Possible values:
```
Comment (veewee) [inline doc/tasks/paratest.md]: The indentation is a bit off. Therefor it doesn't display a list

File: doc/tasks/paratest.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+*Default: null*
+
+Runner, WrapperRunner or SqliteRunner. (Default: Runner)
+
+**debugger**
+
+*Default: null*
+
+Debugger settings for code coverage.
```
Comment (veewee) [inline doc/tasks/paratest.md]: Can you specify the settings here. Unclear what should be in there from the documentation.

File: src/Task/Paratest.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        $coverageEnabled = function ($config) {
+            return !empty($config['coverage-xml'])
+                || !empty($config['coverage-html'])
+                || !empty($config['log-junit']);
+        };
+        if ($coverageEnabled($config) && !empty($config['debugger'])) {
+            $bin = $config['debugger']['bin'];
+            $args = $config['debugger']['args'];
+            $arguments = new ProcessArgumentsCollection(array_merge([$bin], $args, $arguments->toArray()));
```
Comment (veewee) [inline src/Task/Paratest.php]: don't you need to `...$args, ...$arguments->toArray()` ?

File: src/Task/Paratest.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        $arguments->addOptionalArgument('-c %s', $config['configuration']);
+        $arguments->addOptionalArgument('--phpunit %s', $config['phpunit']);
+        $arguments->addOptionalArgument('--runner %s', $config['runner']);
+        $arguments->addOptionalArgument('--coverage-xml %s', $config['coverage-xml']);
+        $arguments->addOptionalArgument('--coverage-html %s', $config['coverage-html']);
+        $arguments->addOptionalArgument('--log-junit %s', $config['log-junit']);
+        $arguments->addOptionalArgument('--testsuite %s', $config['testsuite']);
+        $arguments->addOptionalCommaSeparatedArgument('--group=%s', $config['group']);
+
+        $coverageEnabled = function ($config) {
```
Comment (veewee) [inline src/Task/Paratest.php]: might also be just a variable or a private function

File: src/Task/Paratest.php (hunk 4 above)
Comment (janvernieuwe) [reply src/Task/Paratest.php]: I don't think so? The array merge should handle it to become a single array, and that is what the constructor expects.

File: doc/tasks/paratest.md (hunk 3 above)
Comment (veewee) [reply doc/tasks/paratest.md]: @janvernieuwe  can you add an example?

**Pull request merged**